### PR TITLE
refactor: Deprecate unused actor methods and types

### PR DESCRIPTION
**Pull request by didier-wenzek**
## Proposed changes

- [x] Deprecate `MessageReceiver::recv_message()`
- [x] Deprecate `enum WrappedInput`
- [x] Deprecate `enum RuntimeEvent` (from the public API - still used internally)
- [x] Deprecate `trait ServiceConsumerExt` and the `with_probe()` test helper
- [x] Deprecate `trait ServiceProviderExt` and the `new_client_box()` test helper 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/core/tedge_actors/src/servers/message_boxes.rs [hunk 1]
```diff
@@ -177,7 +180,9 @@ mod tests {
     #[tokio::test]
     async fn only_processes_messages_up_to_max_concurrency() {
         let mut builder = ServerMessageBoxBuilder::new("ConcurrentServerMessageBoxTest", 16);
-        let mut test_box = builder.new_client_box();
+        let mut test_box = SimpleMessageBoxBuilder::new("Test box", 16);
+        test_box.connect_sink(NoConfig, &builder.connect_client(test_box.get_sender()));
+        let mut test_box = test_box.build();
```
Comment (albinsuresh) [inline crates/core/tedge_actors/src/servers/message_boxes.rs:185]: Why not the former variant itself, which is the same but simpler?
```suggestion
    let mut test_box = builder.new_client_box();
```

File: crates/core/tedge_actors/src/servers/message_boxes.rs [hunk 2]
```diff
@@ -281,7 +286,12 @@ mod tests {
             }
 
             fn client_box(&mut self) -> SimpleMessageBox<i32, i32> {
-                self.box_builder.new_client_box()
+                let mut test_box = SimpleMessageBoxBuilder::new("Test box", 16);
+                test_box.connect_sink(
+                    NoConfig,
+                    &self.box_builder.connect_client(test_box.get_sender()),
+                );
+                test_box.build()
```
Comment (albinsuresh) [inline crates/core/tedge_actors/src/servers/message_boxes.rs:294]: Same comment as above.

File: crates/core/tedge_actors/src/servers/mod.rs [hunk 3]
```diff
@@ -151,6 +156,9 @@ pub trait Service<Request: Message, Response: Message>:
     /// The client provides a `response_sender` on which it wants to response to be sent to.
     /// In exchange the client is returned a request sender on which its requests will have to be sent.
     fn connect_client(&mut self, response_sender: DynSender<Response>) -> DynSender<Request>;
+
+    /// Create a simple message box connected to a server box under construction.
+    fn new_client_box(&mut self) -> SimpleMessageBox<Response, Request>;
```
Comment (albinsuresh) [inline crates/core/tedge_actors/src/servers/mod.rs:161]: Wondering if this function should this be hidden behind a test feature flag to make the intent clear?

**Review by albinsuresh — APPROVED** [review]

File: crates/core/tedge_actors/src/servers/message_boxes.rs (hunk 1 above)
Comment (didier-wenzek) [reply crates/core/tedge_actors/src/servers/message_boxes.rs:185]: This is a residue of my attempt to fully remove the `new_client_box()` test helper.

Reverted.

File: crates/core/tedge_actors/src/servers/message_boxes.rs (hunk 2 above)
Comment (didier-wenzek) [reply crates/core/tedge_actors/src/servers/message_boxes.rs:294]: This is a residue of my attempt to fully remove the `new_client_box()` test helper.

Reverted.

File: crates/core/tedge_actors/src/servers/mod.rs (hunk 3 above)
Comment (didier-wenzek) [reply crates/core/tedge_actors/src/servers/mod.rs:161]: Hidden behind `#[cfg(feature = "test-helpers")]`

**Pull request merged**